### PR TITLE
Complete refactor

### PR DESCRIPTION
**Pull request by jlacomis**
That's right, I'm refactoring all of this stuff. For fun.

### CODE REVIEW

File: src/search.ml [hunk 1]
```diff
@@ -1062,13 +1062,15 @@ let ww_adaptive_1 (original : ('a,'b) Rep.representation) incoming_pop =
...
-  if(List.length !excluded_edits) > 0 then begin
-    deletes := List.filter
-      (fun (Delete(src),_,_) ->
-      	let app_str = (Printf.sprintf "d(%d)" src) in
-          not (List.mem app_str !excluded_edits))
-      !deletes
-  end ;
+  if (List.length !excluded_edits) > 0 then begin
+      let delete_filter = function
+        | Delete (src), _, _ ->
```
Comment (rvantonder) [inline src/search.ml]: But why not `src` instead of `(src)`. Yes I'm randomly reading your changes.

Comment (jlacomis) [reply src/search.ml]: Because give me like... a second to deal with this nightmare

**Comment by weimer** [comment]
More power to you two! It is established lore, and likely even true, that all of the questionable code in GenProg dates back to me. As a result, you have my apologies for whatever evil you find in there. Bonus points if you like "functional" programming in which while loops terminate based on the values of mutable reference variables. :-)

**Comment by rvantonder** [comment]
To be fair @weimer if people are still using and fixing your code years down the line, I believe the appropriate sentiment is "you win"

**Pull request closed without merging**